pkg/providers/imagefamily/resolver.go

```diff
@@ -146,16 +151,20 @@ func (r *DefaultResolver) FilterInstanceTypesBySystemDisk(ctx context.Context, n
 				availableSystemDisk.AddAvailableSystemDisk(*resource.Value)
 			}
 		}); err != nil {
-			log.FromContext(ctx).Error(err, "describe available system disk failed")
-			continue
+			errs[i] = err
+			return
 		}
-
 		if availableSystemDisk.Compatible(expectDiskCategories) {
+			resultMutex.Lock()
 			result = append(result, instanceTypes[i])
+			resultMutex.Unlock()
 		} else {
-			log.FromContext(ctx).V(1).Info("%s is not compatible with NodeClass %s SystemDisk %v", instanceType, nodeClass.Name, expectDiskCategories)
+			errs[i] = fmt.Errorf("%s is not compatible with NodeClass %s SystemDisk %v", instanceType.Name, nodeClass.Name, expectDiskCategories)
 		}
 		r.cache.SetDefault(instanceType.Name, availableSystemDisk)
```

Comment:
Will the disk be added to the cache even if it's not compatible with expected?

Reply:
If not compatible, the availableSystemDisk will be empty.
If not set this, the request needs to be done every time. So we should put the key/empty value into the cache.

Reply:
BTW, can you help review the PR if you are available? @jxs1211 
We are very looking forward to your reivew suggestions.